src/extension.ts

```diff
@@ -1,51 +1,81 @@
 import * as vscode from 'vscode';
-import { TabInputCustom, TabInputNotebook, TabInputNotebookDiff, TabInputTerminal, TabInputText, TabInputTextDiff, TabInputWebview, Uri } from 'vscode';
+import {
+  TabInputCustom,
+  TabInputNotebook,
+  TabInputNotebookDiff,
+  TabInputTerminal,
+  TabInputText,
+  TabInputTextDiff,
+  TabInputWebview,
+  Uri,
+} from 'vscode';
 
-type TabContentData = TabInputText | TabInputTextDiff | TabInputCustom | TabInputWebview | TabInputNotebook | TabInputNotebookDiff | TabInputTerminal | unknown;
+type TabContentData =
+  | TabInputText
+  | TabInputTextDiff
+  | TabInputCustom
+  | TabInputWebview
+  | TabInputNotebook
+  | TabInputNotebookDiff
+  | TabInputTerminal
+  | unknown;
+
+let changeTabs = (withNumber: boolean) => async () => {
+  // open a selector for all currently open tabs
+  const tabs: TabWithUri[] = vscode.window.tabGroups.all.flatMap((tabGroup) => {
+    return tabGroup.tabs.flatMap((tab) => {
+      if (isInputWithUri(tab.input)) {
+        return [{ label: tab.label, uri: tab.input.uri }];
+      } else {
+        return [];
+      }
+    });
+  });
+  const selected: string | undefined = await vscode.window.showQuickPick(
+    tabs.map((t, i) => ((withNumber && i.toString() + ' | ') || '') + t.label),
```

Comment:
might make sense to make this start at 1 since keyboards start at 1 on the left.